modules/ROOT/pages/dataweave-type-selector.adoc

```diff
@@ -0,0 +1,209 @@
+= Type Selector
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+The type selector allows the use of the `.` selector over xref:dataweave-type-system.adoc#composite-types[types] which enables you to easily declare types from existing ones. This feature can also be combined with the new xref:dataweave-selectors-reusing-types.adoc[module loader feature] to load files and translate declarations inside your file into DataWeave type directives which can be accessed in the same way as types from any other module:
+
+Example: DataWeave script
+[source,dataweave,linenums]
+----
+%dw 2.0
+---
+type User = {name: String}
+type NameType = User.name
+var name: NameType = "Seba"
+---
+typeOf(name)
+----
+
+Output 
+----
+"String"
+----
+
+== Examples on How to Use Type Selector
+
+The following example describes how type selection is useful for declaring complex types from other types already declared:
```

Comment:
```suggestion
The following example uses the dot selector to declare a new type `Address` from one key in the complex type `User`.
```

Reply:
Suggesting this change bc I think we're declaring the new (less complex) type from the complex type rather than the other way around, and we're showing how you can use the dot/type selector do that. 

Please confirm with dev. Thx

Reply:
Dev's ok with it.